Wikipedia/View Controllers/WebView/WebViewController.m

```diff
@@ -341,17 +341,29 @@ - (CGFloat)clientBoundingRectVerticalOffset {
 }
 
 - (void)scrollToFragment:(NSString*)fragment {
-    if (fragment.length == 0) {
-        // No section so scroll to top. (Used when "Introduction" is selected.)
-        [self.webView.scrollView scrollRectToVisible:CGRectMake(0, 1, 1, 1) animated:NO];
+// Temp fix! See note below about bug with animated scroll approach...
+    if (fragment) {
+        NSString* jumpToThis = [fragment copy];
+        [[self.webView wmf_javascriptContext][@"scrollToFragment"] performSelector:@selector(callWithArguments:) withObject:@[jumpToThis] afterDelay:0.1];
     } else {
-        CGRect r = [self.webView getScreenRectForHtmlElementWithId:fragment];
-        if (!CGRectIsNull(r)) {
-            CGPoint elementOrigin =
-                CGPointMake(self.webView.scrollView.contentOffset.x,
-                            self.webView.scrollView.contentOffset.y + r.origin.y + [self clientBoundingRectVerticalOffset]);
-            [self.webView.scrollView setContentOffset:elementOrigin animated:YES];
-        }
+        [self.webView.scrollView scrollRectToVisible:CGRectMake(0, 1, 1, 1) animated:NO];
+    }
+
+    return;
+
+// Bug with animated scrolling method below:
+// - scroll to bottom of read more items at bottom of article
+// - open TOC and select a section above "Read more"
+// - doesn't work
+// I think it's because clientBoundingRectVerticalOffset doesn't return anything if we've scrolled the web view completely up off the screen for some reason.
```

Comment:
the issue doesn't appear to be w/ `clientBoudningRectVerticalOffset`, but that `getClientBoundingRect()` returns _very_ large negative numbers when invoked while the browser view is offscreen:

```
> document.getElementById("Birth_and_childhood").getBoundingClientRect()
< ClientRect = $11
bottom: -33552278
height: 43.1875
left: -33554416
right: -33554072
top: -33552322
width: 343
```
Where the scroll view's content size is only 49710, and `clientBoundingRectVerticalOffset` returns 0:

```
(lldb) p (CGFloat) [0x7fcf464570e0 clientBoundingRectVerticalOffset]
(CGFloat) $7 = 0
```

However, I would argue that this is an edge case, and would prefer if we scrolled when the rect is valid, otherwise falling back to setting `location.hash`.  In this case, "valid" would mean that the `elementOrigin` we calculate on line 363 is w/in the bounds of the scrollview's content.